src/helpers.c

```diff
@@ -1170,3 +1170,87 @@ bool check_mix_exists(void)
 	free_string(&fullpath);
 	return ret;
 }
+
+/* Iterate the file list and remove from the file system each file/directory */
+void remove_files_from_fs(struct list *files)
+{
+	struct list *iter = NULL;
+	struct file *file = NULL;
+	char *fullfile = NULL;
+	int total = list_len(files);
+	int deleted = total;
+	int count = 0;
+
+	iter = list_head(files);
+	while (iter) {
+		file = iter->data;
+		iter = iter->next;
+		string_or_die(&fullfile, "%s/%s", globals.path_prefix, file->filename);
+		if (swupd_rm(fullfile) == -1) {
+			/* if a -1 is returned it means there was an issue deleting the
+			 * file or directory, in that case decrease the counter of deleted
+			 * files.
+			 * Note: If a file didn't exist it will still be counted as deleted,
+			 * this is a limitation */
+			deleted--;
+		}
+		free_string(&fullfile);
+		count++;
+		progress_report(count, total);
+	}
+	info("Total deleted files: %i\n", deleted);
+}
+
+/*
+ * free all files found in files1 that happens to be duplicated on files2
+ *
+ * Is _mandatory_ to have both files lists ordered by filename before calling
+ * this function
+ *
+ * This function removes all duplicated files matching those in files2 from
+ * files1, but it doesn't free the data.
+ *
+ */
+void deduplicate_files(struct list **files1, struct list *files2)
```

Comment:
same here. this could be a generic list function.

Reply:
You can use the same function that is proposed on #1041 here

Reply:
done, we'll need to merge PR #1041 first then for this to work